src/devices/Ft232H/Ft232HGpio.cs

```diff
@@ -0,0 +1,197 @@
+
+        /// <inheritdoc/>
+        protected override void OpenPin(int pinNumber)
+        {
+            if ((pinNumber < 0) || (pinNumber >= PinCount))
+            {
+                throw new ArgumentException($"Pin number can only be between 0 and {PinCount - 1}");
+            }
+
+            if (DeviceInformation.IsI2cMode && (
+                (pinNumber >= 0) && (pinNumber <= 2)))
+            {
+                throw new ArgumentException($"Can't open pin 0, 1 or 2 while I2C mode is on");
+            }
+
+            if (DeviceInformation.IsSpiMode && (
+                (pinNumber >= 0) && (pinNumber <= 2)))
+            {
+                throw new ArgumentException($"Can't open pin 0, 1 or 2 while SPI mode is on");
+            }
+
+            if (DeviceInformation._connectionSettings.Where(m => m.ChipSelectLine == pinNumber).Any())
+            {
+                throw new ArgumentException($"Pin already open or used as Chip Select");
+            }
+
+            DeviceInformation._PinOpen[pinNumber] = true;
+        }
+
+        /// <inheritdoc/>
+        protected override void ClosePin(int pinNumber)
+        {
+            DeviceInformation._PinOpen[pinNumber] = false;
+        }
+
+        /// <inheritdoc/>
+        protected override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            if (pinNumber < 8)
+            {
+                if (mode != PinMode.Output)
+                {
+                    byte mask = 0xFF;
+                    mask &= (byte)(~(1 << pinNumber));
+                    DeviceInformation._gpioLowDir &= mask;
+                }
+                else
+                {
+                    DeviceInformation._gpioLowDir |= (byte)(1 << pinNumber);
+                }
+
+                DeviceInformation.SetGpioValuesLow();
+            }
+            else
+            {
+                if (mode != PinMode.Output)
+                {
+                    byte mask = 0xFF;
+                    mask &= (byte)(~(1 << (pinNumber - 8)));
+                    DeviceInformation._gpioHighDir &= mask;
+                }
+                else
+                {
+                    DeviceInformation._gpioHighDir |= (byte)(1 << (pinNumber - 8));
+                }
+
+                DeviceInformation.SetGpioValuesHigh();
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override PinMode GetPinMode(int pinNumber)
+        {
+            if (pinNumber < 8)
+            {
+                return ((DeviceInformation._gpioLowDir >> pinNumber) & 0x01) == 0x01 ? PinMode.Output : PinMode.Input;
+            }
+
+            return ((DeviceInformation._gpioHighDir >> (pinNumber - 8)) & 0x01) == 0x01 ? PinMode.Output : PinMode.Input;
+        }
+
+        /// <inheritdoc/>
+        protected override bool IsPinModeSupported(int pinNumber, PinMode mode)
+        {
+            if ((mode == PinMode.InputPullDown) || (mode == PinMode.InputPullUp))
```

Comment:
this is equivalent to `return !((mode == PinMode.InputPullDown) || (mode == PinMode.InputPullUp));`

I'd suggest to do a positive condition though, i.e.: `return mode == PinMode.Input || mode == PinMode.Output;`